### PR TITLE
[Merged by Bors] - Serve Bellatrix preset in BN API

### LINKED ISSUE #2638: /eth/v1/config/spec does not return Altair constants
## Description

The `/eth/v1/config/spec` endpoint does not return all Altair constants.

## Version

1.5.2

## Present Behaviour

Not all Altair constants not returned (although some are, for example `SYNC_COMMITTEE_SIZE`).

## Expected Behaviour

Altair constants should be returned:
- `DOMAIN_CONTRIBUTION_AND_PROOF`
- `DOMAIN_SYNC_COMMITTEE`
- `DOMAIN_SYNC_COMMITTEE_SELECTION_PROOF`
- `SYNC_COMMITTEE_SUBNET_COUNT`
- `TARGET_AGGREGATORS_PER_SYNC_SUBCOMMITTEE`

## Steps to resolve

Query `/eth/v1/config/spec` and examine output.

### LINKED ISSUE #3388: No Bellatrix preset in `/eth/v1/config/spec`
## Description

There's a `TODO(merge)` here which prevents Lighthouse from serving the Bellatrix presets on `/eth/v1/config/spec`:

https://github.com/sigp/lighthouse/blob/21dec6f603edd53ad8d2372941c77ab8098d3b5a/consensus/types/src/config_and_preset.rs#L18-L20

We should try re-enabling that preset and possibly removing all of the other defaults added in https://github.com/sigp/lighthouse/pull/2934.

We should be careful not to break compatibility with any recent versions of Lighthouse or the other clients. I _think_ the VC has become more liberal in what it will accept since that old compat PR was made, but we need to check.

Having dynamic logic to only serve the Bellatrix preset if the Bellatrix epoch is set is another (potentially higher effort) option

### PR DESCRIPTION
**Pull request by michaelsproul**
## Issue Addressed

Resolves #3388
Resolves #2638

## Proposed Changes

- Return the `BellatrixPreset` on `/eth/v1/config/spec` by default.
- Allow users to opt out of this by providing `--http-spec-fork=altair` (unless there's a Bellatrix fork epoch set).
- Add the Altair constants from #2638 and make serving the constants non-optional (the `http-disable-legacy-spec` flag is deprecated).
- Modify the VC to only read the `Config` and not to log extra fields. This prevents it from having to muck around parsing the `ConfigAndPreset` fields it doesn't need.

## Additional Info

This change is backwards-compatible for the VC and the BN, but is marked as a breaking change for the removal of `--http-disable-legacy-spec`.

I tried making `Config` a `superstruct` too, but getting the automatic decoding to work was a huge pain and was going to require a lot of hacks, so I gave up in favour of keeping the default-based approach we have now.

### CODE REVIEW

File: beacon_node/src/config.rs [hunk 1]
```diff
@@ -116,7 +116,14 @@ pub fn get_config<E: EthSpec>(
     }
 
     if cli_args.is_present("http-disable-legacy-spec") {
-        client_config.http_api.serve_legacy_spec = false;
+        warn!(
+            log,
+            "The flag --http-disable-legacy-spec is deprecated and will be removed"
```
Comment (paulhauner) [inline beacon_node/src/config.rs:121]: Perhaps you want to point users towards `--http-spec-fork`? I'm easy, just a suggestion :)

**Review by paulhauner — APPROVED** [review]
LGTM! This looks really nice, it's a shame that API endpoint has become so complex!

File: beacon_node/src/config.rs (hunk 1 above)
Comment (michaelsproul) [reply beacon_node/src/config.rs:121]: they're pretty much unrelated (mutually exclusive sets of feilds) so I reckon we leave it

**Pull request closed without merging**